### PR TITLE
#153 queries show the actual query when we click on test sql query, also keep showing the error message somewhere

### PR DESCRIPTION
**Pull request by Mai21**
Closes 153

# 👩🏻‍💻 What does this PR do?
 
- Added "Generate query" in the side panel 
- Added a permanent alert panel instead of a snack bar 
- Added an information panel for the number of rows

When the query returns an error
Screenshot 2023-11-01 at 11 08 48 AM

When the query returns the result
Screenshot 2023-11-01 at 11 09 05 AM

# 🧪 How has/should this change been tested?

## 📃 Documentation

_No user facing changes_

or

- [ ] Functional change 1
- [ ] Functional change 2

### CODE REVIEW

File: frontend/packages/common/src/ui/components/panels/InfoPanel.tsx [hunk 1]
```diff
@@ -10,8 +10,6 @@ export const InfoPanel = ({ message }: { message: string }) => (
     alignItems="center"
     sx={{
       backgroundColor: 'gray.pale',
-      borderRadius: '10px',
```
Comment (jmbrunskill) [inline frontend/packages/common/src/ui/components/panels/InfoPanel.tsx:13]: Normally I'd be careful about changing a common components settings like this.
I have checked it's not used anywhere else though, so in this case it's fine.

File: frontend/packages/system/src/Queries/NotificationQueries/DetailEdit.tsx [hunk 2]
```diff
@@ -101,9 +118,18 @@ export const DetailEdit = () => {
       </AppBarContentPortal>
       {/* Sql Results table */}
       <Box sx={{ width: '100%', display: 'flex', flexDirection: 'column' }}>
-        <Box sx={{ flex: '1', overflow: 'auto' }}>
+        {queryError && (<AlertPanel message ={queryError}/>)}
+        {(generatedQuery && !queryError) && (<InfoPanel message ={'Result: ' + sqlResults.length.toString() + ' rows'}/>)}
```
Comment (jmbrunskill) [inline frontend/packages/system/src/Queries/NotificationQueries/DetailEdit.tsx:122]: Maybe should create a translation for this? Not a priority though...

File: frontend/packages/system/src/Queries/NotificationQueries/QueryEditor.tsx [hunk 3]
```diff
@@ -257,7 +257,6 @@ export const QueryEditor = ({
             onClick={() => {
               onSave(draft).catch(err => {
                 console.error(err);
-                error(err)();
```
Comment (jmbrunskill) [inline frontend/packages/system/src/Queries/NotificationQueries/QueryEditor.tsx:260]: Why remove this?

Comment (jmbrunskill) [reply frontend/packages/system/src/Queries/NotificationQueries/QueryEditor.tsx:260]: There could be an error saving which would be good to show to the user if that happens...

**Review by jmbrunskill — APPROVED** [review]
Looking good. Have a few suggestions for improvements to the code...

File: frontend/packages/common/src/ui/components/panels/InfoPanel.tsx (hunk 1 above)
Comment (Mai21) [reply frontend/packages/common/src/ui/components/panels/InfoPanel.tsx:13]: I understand it:) It is not used, so I changed it. But if it shouldn't be changed, I will create a new component for this!

File: backend/graphql/datasource/src/lib.rs [hunk 4]
```diff
@@ -33,8 +36,8 @@ impl DatasourceQueries {
         // TODO some kind of query validation?
 
         // Query datasource service and return result
-        match datasource_service.run_sql_query(sql_query) {
-            Ok(result) => Ok(result),
+        match datasource_service.run_sql_query_with_parameters(sql_query, "".to_string()) {
```
[the remark this answers is not in the document]
Comment (Mai21) [reply backend/graphql/datasource/src/lib.rs:39]: @jmbrunskill It seems not being used but I think I changed it accidentally... I'll return this to the previous one

```Rust
match datasource_service.run_sql_query(sql_query) {
```

File: frontend/packages/system/src/Queries/NotificationQueries/QueryEditor.tsx (hunk 3 above)
Comment (Mai21) [reply frontend/packages/system/src/Queries/NotificationQueries/QueryEditor.tsx:260]: The reason why I remove this is I removed `error` here. This is for using snackbar. 
https://github.com/openmsupply/notify/blob/be0151fb95dee12c4a1042e913e668f449453577/frontend/packages/system/src/Queries/NotificationQueries/QueryEditor.tsx#L61 

But for saving in the QueryEditor, I agree with you, we better have this error catch. I'll get it back.

File: frontend/packages/system/src/Queries/NotificationQueries/DetailEdit.tsx (hunk 2 above)
Comment (Mai21) [reply frontend/packages/system/src/Queries/NotificationQueries/DetailEdit.tsx:122]: Done

**Review by jmbrunskill — APPROVED** [review]
Looks good thanks for the updates!

File: frontend/packages/common/src/intl/locales/en/system.json [hunk 5]
```diff
@@ -73,6 +73,7 @@
   "messages.password-weak": "Weak",
   "messages.recipients-added_other": "{{count}} recipients added to list",
   "messages.recipients-added_one": "{{count}} recipient added to list",
+  "messages.query-result-count": "Result: {{number}} rows",
```
[the remark this answers is not in the document]
Comment (Mai21) [reply frontend/packages/common/src/intl/locales/en/system.json]: @jmbrunskill I was wondering how to use `{{count}}` then decided to go with `{{number}}`. 
I fixed it but when the result is 0, it also returns with `0 rows`. Is there any way that we can return 0 row, 1 row or 2 rows?

[the remark this answers is not in the document]
Comment (jmbrunskill) [reply frontend/packages/common/src/intl/locales/en/system.json]: I think how it's working is correct `0 rows` is the right way to say it.

https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/z/zero-zeros

[the remark this answers is not in the document]
Comment (jmbrunskill) [reply frontend/packages/common/src/intl/locales/en/system.json]: Here's the library docs
https://www.i18next.com/translation-function/plurals#singular-plural
https://www.i18next.com/translation-function/plurals#languages-with-multiple-plurals

**Pull request merged**